### ISSUE TITLE
Color picker is always late by one click

### ISSUE THREAD

**Issue by targos** [cheminfo/nmrium#1532]
http://localhost:3000/#/SamplesDashboard/dutmznbn18s/Viewdatacytisinedatacytisine1Hjson
https://github.com/cheminfo/nmrium/runs/6287161946?check_suite_focus=true

https://user-images.githubusercontent.com/2352663/166655397-45ac1949-2b95-4423-bf66-f40821c8a979.mp4

This is probably related to the following React warning:
```
react_devtools_backend.js:3973 
        
       Warning: componentWillReceiveProps has been renamed, and is not recommended for use. See https://reactjs.org/link/unsafe-component-lifecycles for details.

* Move data fetching code or side effects to componentDidUpdate.
* If you're updating state whenever props change, refactor your code to use memoization techniques or move it to static getDerivedStateFromProps. Learn more at: https://reactjs.org/link/derived-state
* Rename componentWillReceiveProps to UNSAFE_componentWillReceiveProps to suppress this warning in non-strict mode. In React 18.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run `npx react-codemod rename-unsafe-lifecycles` in your project source folder.

Please update the following components: ColorPicker2, EditableInput2
```

**Comment by targos** [comment]
BTW it would also be nice if we could fix the color property when it's set to white, because as we see in the test, it becomes `#ffffffff`, wich is invalid CSS color.

**Comment by hamed-musallam** [comment]
@targos 

it comes from a component we do not own, it is from https://github.com/casesandberg/react-color, I think the last release was 2 years ago, what do you suggest? raising an issue on their repository? or looking for another component?

**Comment by targos** [comment]
I suggest we try to find another more maintained component.

**Comment by hamed-musallam** [comment]
@targos @lpatiny 
what do you think about this colour picker package https://github.com/omgovich/react-colorful?

**Comment by targos** [comment]
LGTM

**Comment by hamed-musallam** [comment]
I raised an issue at their repository because we have a problem if we include it under root shadow

https://github.com/omgovich/react-colorful/issues/180

**Comment by targos** [comment]
I discussed with Luc, and we think that the best course of action will be to take over the code from react-color and fix it ourselves.
So the idea would be to copy the files needed for the component (I think it's Sketch?) from https://github.com/casesandberg/react-color into `analysis-ui-components`. Copy the license file as well for attribution.
Then we will be able to fix it for the new React version and maybe even improve it in the future.
What do you think? Can you work on this?

**Comment by hamed-musallam** [comment]
closed by [#736 ](https://github.com/cheminfo/nmrium/commit/09a910d47a990df3b07c0c739ad1f540591b77b9)
